### PR TITLE
add screenshots

### PR DESCRIPTION
**Pull request by tobes**
This implements the ideas in #667.  A big shout out to @lasers for the original idea, support, testing etc

We create a new modules.md https://github.com/tobes/py3status/blob/screenshots/doc/modules.md that includes both 'screenshots' of modules and `<details>` blocks so it is more compact and user friendly.  I think the modules/README.md is actually a useful document as it is because it can be searched etc and so this PR creates a new page with screenshots (I think we should link to the new page on the default README.md).

screenshot data is kept in the module docstrings as I think this makes the most sense.

There is still some work that needs doing.

* document it.
* add links on README.md etc
* some modules lack screenshots.
* add tests to ensure modules have screenshots.

to use either `py3status screenshots` creates screenshots or `py3status docstring update ...` and screenshots/modules.md will also be created.

We need python modules `cairo` `pango` `pangocairo` to create/update screenshots.  If these are missing we will just build `doc/modules.md` using existing screenshots

@ultrabug sorry this is so big.  Let me know if it needs breaking up etc. or any other changes you feel are needed.

### CODE REVIEW

File: py3status/modules/group.py [hunk 1]
```diff
@@ -78,6 +78,9 @@
 ```
 
 @author tobes
+
+SAMPLE OUTPUT
+{'full_text': 'output of another module'}
```
Comment (lasers) [inline py3status/modules/group.py]: I think `module 1` would look clean for screenshot... similar to frame's `module 1|module 2|module 3`

**Comment by tobes** [comment]
So I've updated this with @lasers fixups/examples

other changes:

* we can now show urgent eg for github (and background - not that it is used anywhere)

* All screenshots for a module share the same 'color scheme' it was getting a bit too colorful and this uniformity makes things clearer.

**Comment by lasers** [comment]
>I didn't take the (fixme) bits because I think that not having one is better than a "we don't have one". I also made a few minor tweeks.

No problem with this. I explicitly put it there to make it clear. Hope somebody who uses them will take time to give us the missing (fixme) information instead of just skipping through not realizing.

P.S. It turns out that you can't use py3status (py3) on this... only py3status2 (py2). 😈
I think urgent should be orange... a mix of red/yellow.... a mix of error/warning/informational color schemes. I want to add urgent for imap too, but need to wait for allow_urgent dealio  to work out first. Will look at new changes.

**Comment by tobes** [comment]
> P.S. It turns out that you can't use py3status (py3) on this... only py3status2 (py2). 😈

Is possible I don't have the cairo libs installed for python3 I'll fix it.


> I think urgent should be orange... a mix of red/yellow.... a mix of error/warning/informational color schemes. I want to add urgent for imap too, but need to wait for allow_urgent dealio to work out first. Will look at new changes.

`#900000` is the default so that's why I've gone with it.

**Comment by lasers** [comment]
DNT new defaults? Just DNT (green/red) instead of OFF/ON (green/red)? Consistent w/ DPMS, XSCR, BT, etc....

There actually are two icinga2 modules, one simple. one w/ more colors. I think the author improved icinga2 on his repostiory sometimes after py3status got the simple module. https://github.com/nazco/i3status-modules  

IMAP will have white "Mail: 0" only if it's zero. Maybe we should add that along with IMAP urgent (if it goes in).

whatismyip, should we omit the second ip? just "Russia"? More clean for screenshot.

Overall, it looks ready/great. Will head out. IRC later. Ty for fixing urgent, cheers. 👍

**Comment by tobes** [comment]
I'm going to move this into another branch I'm working on

**Pull request closed without merging**